compiler/src/main/java/dagger/internal/codegen/BindingGraph.java

```diff
@@ -950,7 +950,8 @@ void resolve(BindingKey bindingKey) {
          * 2. If there are any explicit bindings in this component, they may conflict with those in
          *    the supercomponent, so resolve them here so that conflicts can be caught.
          */
-        if (getPreviouslyResolvedBindings(bindingKey).isPresent()) {
+        if (getPreviouslyResolvedBindings(bindingKey).isPresent()
+            && bindingKey.kind().equals(BindingKey.Kind.CONTRIBUTION)) {
```

Comment:
I'm not sure this is the right fix, given my comment below about the duplicated members injector. My hunch is that there's something not working right with the call to `getLocalExplicitBindings()` when used on a `Key` for members injection. Maybe that's always returning empty?